### PR TITLE
feat(core,console): social connector targets

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

In "sign in experience setup", as for connectors, we don't care about `platform`, we are actually sorting and setting connectors by `target`.

So change `socialConnectorIds` to `socialConnectorTargets`, when requesting settings (in UI), map targets to related connectors.

## Linear Issue Reference

LOG-2438

## Testing

Pass all tests, and local tested:

截屏2022-05-17 上午10 39 36

### CODE REVIEW

File: packages/console/src/pages/SignInExperience/utilities.ts [hunk 1]
```diff
@@ -81,11 +81,15 @@ export const compareSignInMethods = (
...
+  if (before.socialSignInConnectorTargets.length !== after.socialSignInConnectorTargets.length) {
     return false;
   }
 
-  if (before.socialSignInConnectorIds.some((id) => !after.socialSignInConnectorIds.includes(id))) {
+  if (
+    before.socialSignInConnectorTargets.some(
+      (target) => !after.socialSignInConnectorTargets.includes(target)
+    )
+  ) {
```
Comment (darcyYe) [inline packages/console/src/pages/SignInExperience/utilities.ts:92]: Want to make sure this method cares only about the existence of targets not the order of targets?

Comment (wangsijie) [reply packages/console/src/pages/SignInExperience/utilities.ts:92]: YES

File: packages/console/src/hooks/use-connector-groups.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            enabled: item.enabled,
+            connectors: [item],
+          },
+        ];
+      }
+
+      return previous.map((group, index) => {
+        if (index !== groupIndex) {
+          return group;
+        }
```
Comment (darcyYe) [inline packages/console/src/hooks/use-connector-groups.ts:36]: will this ever happen?

File: packages/core/src/routes/session.ts [hunk 3]
```diff
@@ -581,12 +582,18 @@ export default function sessionRoutes<T extends AnonymousRouter>(router: T, prov
...
+      Array<ConnectorMetadata & { id: string }>
+    >((previous, connectorTarget) => {
+      const connectors = connectorInstances.filter(
+        ({ metadata: { target } }) => target === connectorTarget
+      );
+
+      return [
+        ...previous,
+        ...connectors.map(({ metadata, connector: { id } }) => ({ ...metadata, id })),
+      ];
```
Comment (IceHe) [inline packages/core/src/routes/session.ts:595]: nit: better add UTs for multiple connectors with the same `target` in one group.

**Review by IceHe — APPROVED** [review]
Overall LGTM.

File: packages/console/src/hooks/use-connector-groups.ts (hunk 2 above)
Comment (wangsijie) [reply packages/console/src/hooks/use-connector-groups.ts:36]: YES, only push new connectors to the specific index.

File: packages/core/src/routes/session.ts (hunk 3 above)
Comment (wangsijie) [reply packages/core/src/routes/session.ts:595]: good point, updated

**Pull request merged**